packages/utils/test/address-sort.spec.ts

```diff
@@ -2,50 +2,176 @@
+        multiaddr: multiaddr('/ip4/30.0.0.1/tcp/4000/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN/p2p-circuit/p2p/QmcrQZ6RJdpYuGvZqD5QEHAv6qX4BrQLJLQPQUrTrzdcgm'),
+        isCertified: false
+      }
+
+      const addresses = [
+        publicRelay,
+        publicAddress
+      ]
+
+      const sortedAddresses = addresses.sort(circuitRelayAddressesLast)
+      expect(sortedAddresses).to.deep.equal([
+        publicAddress,
+        publicRelay
+      ])
+    })
+  })
+
+  describe('default address sort', () => {
+    it('should sort public, then public relay, then private, then private relay with certified addresses taking priority', () => {
+      const certifiedPublicAddress = {
+        multiaddr: multiaddr('/ip4/30.0.0.1/tcp/4001'),
+        isCertified: true
+      }
+      const publicAddress = {
         multiaddr: multiaddr('/ip4/30.0.0.1/tcp/4000'),
         isCertified: false
-      },
-      {
-        multiaddr: multiaddr('/ip4/31.0.0.1/tcp/4000'),
+      }
+      const certifiedPublicRelay = {
+        multiaddr: multiaddr('/ip4/30.0.0.1/tcp/4001/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN/p2p-circuit/p2p/QmcrQZ6RJdpYuGvZqD5QEHAv6qX4BrQLJLQPQUrTrzdcgm'),
+        isCertified: true
+      }
+      const publicRelay = {
+        multiaddr: multiaddr('/ip4/30.0.0.1/tcp/4000/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN/p2p-circuit/p2p/QmcrQZ6RJdpYuGvZqD5QEHAv6qX4BrQLJLQPQUrTrzdcgm'),
+        isCertified: false
+      }
+      const certifiedPrivateAddress = {
+        multiaddr: multiaddr('/ip4/127.0.0.1/tcp/4001'),
         isCertified: true
       }
-    ]
+      const privateAddress = {
+        multiaddr: multiaddr('/ip4/127.0.0.1/tcp/4000'),
+        isCertified: false
+      }
+      const certifiedPrivateRelay = {
+        multiaddr: multiaddr('/ip4/127.0.0.1/tcp/4001/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN/p2p-circuit/p2p/QmcrQZ6RJdpYuGvZqD5QEHAv6qX4BrQLJLQPQUrTrzdcgm'),
+        isCertified: true
+      }
+      const privateRelay = {
+        multiaddr: multiaddr('/ip4/127.0.0.1/tcp/4000/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN/p2p-circuit/p2p/QmcrQZ6RJdpYuGvZqD5QEHAv6qX4BrQLJLQPQUrTrzdcgm'),
+        isCertified: false
+      }
+
+      const addresses = [
+        privateAddress,
+        certifiedPrivateAddress,
+        publicRelay,
+        certifiedPublicRelay,
+        privateRelay,
+        publicAddress,
+        certifiedPublicAddress,
+        certifiedPrivateRelay
+      ].sort(() => {
+        return Math.random() > 0.5 ? -1 : 1
+      })
+
+      const sortedAddresses = addresses.sort(defaultAddressSort)
+      expect(sortedAddresses).to.deep.equal([
+        certifiedPublicAddress,
+        publicAddress,
+        certifiedPublicRelay,
+        publicRelay,
+        certifiedPrivateAddress,
+        privateAddress,
+        certifiedPrivateRelay,
+        privateRelay
+      ])
+    })
+
+    it('should sort WebRTC over relay addresses before relay addresses', () => {
```

Comment:
There is no extra logic - the test is there to ensure we aren't matching WebRTC addresses as Circuit Relay addresses (since they usually have a `p2p-circuit` component).

Reply:
Yes I wasn't seeing where that matching was done but I searched and see it's performed by the [Circuit.exactMatch](https://github.com/multiformats/js-multiaddr-matcher/blob/main/src/index.ts#L572-L586) that's performed [here](https://github.com/libp2p/js-libp2p/pull/2031/files#diff-c13f18cbd49ec253408bbdd970b6bebc76b70024e516cd4f48d837dac276d190R63-R64)